src/v3.ts

```diff
@@ -43,112 +38,98 @@ export default function generateTypesV3(
+          !node.allOf &&
           !node.additionalProperties
         ) {
-          return escape(`{[key: string]: any}`);
+          return `{ [key: string]: any }`;
         }
 
-        // unescape properties if some have been transformed already for nested objects
-        const properties = makeOptional(
-          fromEntries(
-            Object.entries(
-              (node.properties as OpenAPI3SchemaObject["properties"]) || {}
-            ).map(([key, val]) => {
-              if (typeof val === "string") {
-                // try and parse as JSON to remove bad string escapes; otherwise, escape normally
-                try {
-                  return [key, JSON.parse(val)];
-                } catch (err) {
-                  return [key, escape(unescape(val))];
-                }
-              }
-              return [key, val];
-            })
-          ),
-          node.required
-        );
+        let properties = createKeys(node.properties || {}, node.required);
 
         // if additional properties, add to end of properties
         if (node.additionalProperties) {
-          const addlType =
-            typeof node.additionalProperties === "string" &&
-            PRIMITIVES[unescape(node.additionalProperties)];
-          properties[escape("[key: string]")] = escape(addlType || "any");
+          properties += `[key: string]: ${
+            nodeType(node.additionalProperties) || "any"
+          };\n`;
         }
 
         return tsIntersectionOf([
-          // append allOf first
-          ...(node.allOf
-            ? node.allOf.map((item: any) =>
-                isSchemaObj(item)
-                  ? JSON.stringify(makeOptional(item, node.required))
-                  : item
-              )
-            : []),
-          // then properties + additionalProperties
-          ...(Object.keys(properties).length
-            ? [JSON.stringify(properties)]
-            : []),
+          ...(node.allOf ? (node.allOf as any[]).map(transform) : []), // append allOf first
+          ...(properties ? [`{ ${properties} }`] : []), // then properties + additionalProperties
         ]);
       }
+      case "array": {
+        return tsArrayOf(transform(node.items as any));
+      }
+    }
+
+    return "";
+  }
+
+  function createKeys(
+    obj: { [key: string]: any },
+    required?: string[]
+  ): string {
+    let output = "";
 
-      // arrays
-      if (isArrayNode(node) && typeof node.items === "string") {
-        return escape(tsArrayOf(node.items));
+    Object.entries(obj).forEach(([key, value]) => {
+      // 1. JSDoc comment (goes above property)
+      if (value.description) {
+        output += comment(value.description);
       }
 
-      return node; // return by default
-    }
-  );
+      // 2. name (with “?” if optional property)
+      output += `"${key}"${!required || !required.includes(key) ? "?" : ""}: `;
+
+      // 3. open nullable
+      if (value.nullable) {
+        output += "(";
+      }
```

Comment:
cc @antonk52 sorry about doing this for you, but your PR pointed a flaw in the design. And I wanted to make sure I was solving for `nullable` (and it seems to be!).

So this _generates_, and it _looks_ correct, but I actually didn’t add tests in this PR making sure that `| null` works in all cases. Are nullable tests (and potentially fixing some edge cases) something you’re interested in writing in a followup PR? 👍 / 👎 ?

Reply:
No worries man, whatever gets the job done =)

IMHO I think the existing cases covered the nullable fairly well. If you can point out a few extra test cases where the explicit test cases are needed I have no problem sending a pr later on

Reply:
Yeah, fair. Thanks for understanding.

Also, a new version is up on npm with the changes, so I’d love to see if the new `@next` version works for your schemas :)